pkg/guardian/redis_conf_store.go

```diff
@@ -592,6 +915,43 @@ func (rs *RedisConfStore) UpdateCachedConf() {
 		rs.conf.blacklist = fetched.blacklist
 	}
 
+	if fetched.globalRateLimit != nil {
+		rs.conf.globalRateLimit = *fetched.globalRateLimit
+		rs.reporter.CurrentGlobalLimit(rs.conf.globalRateLimit.Spec.Limit)
+	}
+
+	if fetched.globalSettings != nil {
+		rs.conf.globalSettings = *fetched.globalSettings
+		rs.reporter.CurrentReportOnlyMode(rs.conf.globalSettings.Spec.ReportOnly)
+	}
+
+	rs.conf.rateLimitsByName = make(map[string]RateLimitConfig)
+	rs.conf.rateLimitsByURL = make(map[url.URL]RateLimitConfig)
```

Comment:
Given that url.URL is a struct with a number of fields, I'm not sure how I feel about the runtime implication of using it as a map key. 

Also since structs are only equal if every field is equal (including unexported fields) and url.URL contains a pointer field, I believe this will result in unintuitive behavior: https://play.golang.org/p/IGVwrcIWPZO

Reply:
This is definitely very brittle. I honestly just did this as that's how it was being done before (conf.RouteRateLimits, etc) but it should be changed, and it will make even less sense if/when the Guardian interfaces are changed to pass a Request instead of a URL. I'll get on this change now.

Reply:
Fixed by 67b51a7.